packages/test/src/util/browser-tests-runner/driver.js

```diff
@@ -122,15 +133,45 @@ function connect({ viewport = DEFAULT_VIEWPORT, ...options }, capability) {
       }
     })
     .init()
-    .url("")
-    .timeouts("script", timeout)
-    .timeouts("implicit", timeout)
-    .timeouts("page load", timeout)
+    .timeouts("script", idleTimeout)
+    .timeouts("implicit", idleTimeout)
+    .timeouts("page load", idleTimeout)
     .setViewportSize(viewport)
+    .url("")
     .then(() => () => {
-      return driver.executeAsync(function(done) {
-        window.__run_tests__(done);
-      });
+      const endTime = Date.now() + suiteTimeout;
+      const execInterval = Math.min(idleTimeout, DEFAULT_TIMEOUTS.idle) * 0.8;
+      // Some services force a max timeout for async scripts.
+      // Below we restart the async script the waits for the tests to be done every 60 seconds.
+      // The tests are 'complete' once the global '__test_result__' is set on the window.
+      return (function getResults() {
+        if (endTime < Date.now()) {
+          throw new Error(
+            'marko-cli: Test suite timed out, use "wdioOptions.suiteTimeout" to increase the delay (default 10 mins).'
+          );
+        }
+
+        return driver
+          .executeAsync(function(interval, done) {
+            if (window.__test_result__) done(window.__test_result__);
+            else {
+              var timeout = setTimeout(function() {
+                delete window.__test_result__;
+                done();
+              }, interval);
+              Object.defineProperty(window, "__test_result__", {
+                configurable: true,
+                set: function(result) {
+                  clearTimeout(timeout);
+                  done(result);
+                }
+              });
+            }
+          }, execInterval)
+          .then(result => {
+            return result.value ? result : getResults();
```

Comment:
Just to be clear...does this `result` come from the argument passed to `done(...)` above? If so, then if the timeout function gets called, `done()` will get called with no arguments resulting in this code throwing a null reference error on `result.value`.

Reply:
`result` is a webdriver.io result with `result.value` being what was passed to done. Basically this just keeps checking to see if the `executeAsync` was completed by either the test result or the timeout. If it was the timeout (no value) we try to get the results again. This means that as soon as the tests are done it sends the result back.

The reason I had to do this is because services (specifically testing against browserstack) have a hard limit on the duration of a single command (`executeAsync` in this case) which is very low. They do this because most people are not actually running the tests In the browser like we are, but using the driver to run individual tests.